adserver/tasks.py

```diff
@@ -417,3 +420,28 @@ def calculate_publisher_ctrs(days=7):
 
         publisher.sampled_ctr = report.total["ctr"]
         publisher.save()
+
+
+@app.task()
+def notify_of_completed_flights():
+    """Send a notification about flights which completed in the last day."""
+    cutoff = get_ad_day() - datetime.timedelta(days=1)
+    for flight in Flight.objects.filter(live=True).filter():
+        if (
+            flight.clicks_remaining() == 0
+            and flight.views_remaining() == 0
+            and AdImpression.objects.filter(
+                date__gte=cutoff,
+                advertisement__flight=flight,
+            ).exists()
```

Comment:
This is good logic, but I wonder if we should add some state to make sure this happens? If we're confident this will run daily and complete thats good, but if it fails we won't catch it. 

This is fine for a v1, but I think we probably want something on the model defining `alerted` or similar, especially if we're emailing users.

Reply:
I agree completely. If we're emailing users, we need an additional flag.